app/main.py

```diff
@@ -1 +1,43 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __del__(self) -> None:
+        if self.hidden and self in Animal.alive:
+            Animal.alive.remove(self)
```

Comment:
The `__del__` method is not guaranteed to be called for objects that still exist when the interpreter exits. Relying on it for critical cleanup tasks, such as removing an object from a list, may lead to unexpected behavior. Consider using a different mechanism for managing the `alive` list.